lib/ci-config-stack.ts

```diff
@@ -24,10 +24,24 @@ export class CIConfigStack extends Stack {
     constructor(scope: Construct, id: string, props?: StackProps) {
       super(scope, id, props);
 
-      const arnSecret = new Secret(this, 'certificateArn', {});
-      const certContentsSecret = new Secret(this, 'certContents', {});
-      const privateKeySecret = new Secret(this, 'privateKey', {});
-      const redirectUrlSecret = new Secret(this, 'redirectUrl', {});
+      const kmsKey = new Key(this, 'ciKey', {
+        description: 'KMS key for encrypting ci-config secrets',
+        enableKeyRotation: true,
+        // removalPolicy: RemovalPolicy.DESTROY,
```

Comment:
The default retain policy is `RETAIN` for this. do we want to keep that or destroy it went the stack is destroyed?

Reply:
I think we should destroy it and then create one again when the stack is created.

Reply:
If we are making a special purpose encryption key, it should have the same lifecycle associated with it it.  The issue with retain is that it can block re-deployment since the key will already exist.

The reason that the default policy is `RETAIN` on a KMS Key is that if you were to use that key on anything else, it becomes irretrievably lost when the key vanishes, so its a guardrail - I would say `DESTORY` is appropriate for this use case.

Reply:
Agreed. I'll update the PR (if if we are going ahead with this change).
Thanks!